---
title: self 에 대하여
date: 2022-01-18
description: 파이썬, 문법
---

- 클래스 내부에 정의된 함수인 메서드의 첫번째 인자는 인스턴스 즉, 자기자신이다. 
그러기에 self라는 이름으로 쓰고 있고 인자를 아무것도 정의하지 않고 호출하면 에러가 생긴다.  
파이썬의 클래스는 그 자체가 하나의 네임스페이스이기 때문에 인스턴스 생성과 상관없이 클래스 내의 메서드를 직접 호출할 수 있다.  
그러기에 코드 내에 self.forpass(x)가 있었는데 self는 현재 자기자신의 클래스(인스턴스)이기에 그 내부의 
forpass()함수를 호출할 수 있었던 것이다.  

```python
def forpass(self,x):
    ...
def training(self, x, y):
    self.forpass(x)
    ....